### PR TITLE
21169: Infer feature attributes: Added an example CSV parser, fixed a bug that caused iterating over return values to order alphabetically, updated the unknown type defaults

### PR DESCRIPTION
**Pull request by lancegliser**
- Altered the unknown feature return type to `{ type: "continuous", data_type: "number", bounds: { allow_null: true } }`
- Fixed a bug that caused feature attributes to appear in alphabetical instead of numeric order when used in iterations

### CODE REVIEW

File: src/features/sources/CSV.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/*
```
Comment (fulpm) [inline src/features/sources/CSV.ts:1]: I wonder if this file should be renamed to say, `csv.example.ts` instead, so its not confused with a real infer option.

File: src/tests/assets/asteroids-sample.csv [hunk 2]
```diff
@@ -0,0 +1,37 @@
+full_name,a,e,G,i,om,w,q,ad,per_y,data_arc,condition_code,n_obs_used,H,diameter,extent,albedo,rot_per,GM,BV,UB,IR,spec_B,spec_T,neo,pha,moid
```
Comment (fulpm) [inline src/tests/assets/asteroids-sample.csv:1]: We should probably add citations for these datasets.

Comment (fulpm) [reply src/tests/assets/asteroids-sample.csv:1]: Going to add a file in my branch for this, and I can go ahead and include this one too.

File: src/features/sources/CSV.ts (hunk 1 above)
Comment (lancegliser) [reply src/features/sources/CSV.ts]: Moved CSV to examples/CSV

**Review by fulpm — APPROVED** [review]

**Pull request merged**